runtime/src/message_processor.rs

```diff
@@ -500,6 +500,8 @@ impl MessageProcessor {
             // Verify the calling program hasn't misbehaved
             invoke_context.verify_and_update(message, instruction, accounts)?;
 
+            invoke_context.record_instruction(&instruction);
```

Comment:
Does the message in this function have the same account keys and ordering as the original message?

Reply:
Not sure but I don't think it matters because at this point the instruction is compiled